From 5a55b1ce3a4b886db5b7148f58d6bb36d1ca7250 Mon Sep 17 00:00:00 2001
From: timyao <tim.yao@amlogic.com>
Date: Fri, 6 Jun 2014 11:46:53 -0700
Subject: [PATCH 4304/5965] PD #88700-2. Remove is_vpp check based on the log
 captured to avoid release of current display frame CMA buffer.

---
 drivers/amlogic/amports/vh264_4k2k.c | 18 ++++++------------
 1 file changed, 6 insertions(+), 12 deletions(-)

diff --git a/drivers/amlogic/amports/vh264_4k2k.c b/drivers/amlogic/amports/vh264_4k2k.c
index 595ec67cb11a..47a49f66d7a4 100755
--- a/drivers/amlogic/amports/vh264_4k2k.c
+++ b/drivers/amlogic/amports/vh264_4k2k.c
@@ -347,17 +347,14 @@ int init_canvas(int start_addr, long dpb_size, int dpb_number, int mb_width, int
     u32 disp_addr = 0xffffffff;
     bool use_alloc = false;
     int alloc_count = 0;
+    canvas_t cur_canvas;
 
     dpb_addr = start_addr + dpb_size;
 
     mb_total = mb_width * mb_height;
 
-    if (is_vpp_postblend()) {
-        canvas_t cur_canvas;
-
-        canvas_read((READ_VCBUS_REG(VD1_IF0_CANVAS0) & 0xff), &cur_canvas);
-        disp_addr = (cur_canvas.addr + 7) >> 3;
-    }
+    canvas_read((READ_VCBUS_REG(VD1_IF0_CANVAS0) & 0xff), &cur_canvas);
+    disp_addr = (cur_canvas.addr + 7) >> 3;
 
     mutex_lock(&vh264_4k2k_mutex);
     
@@ -1246,6 +1243,7 @@ static int vh264_4k2k_stop(void)
 {
     int i;
     u32 disp_addr = 0xffffffff;
+    canvas_t cur_canvas;
 
     if (stat & STAT_VDEC_RUN) {
         amvdec_stop();
@@ -1281,12 +1279,8 @@ static int vh264_4k2k_stop(void)
     amvdec_disable();
     amvdec2_disable();
 
-    if (is_vpp_postblend()) {
-        canvas_t cur_canvas;
-
-        canvas_read((READ_VCBUS_REG(VD1_IF0_CANVAS0) & 0xff), &cur_canvas);
-        disp_addr = cur_canvas.addr;
-    }
+    canvas_read((READ_VCBUS_REG(VD1_IF0_CANVAS0) & 0xff), &cur_canvas);
+    disp_addr = cur_canvas.addr;
 
     for (i=0; i<ARRAY_SIZE(buffer_spec); i++) {
         if (buffer_spec[i].alloc_pages) {
-- 
2.19.0

